pkg/image/api/validation/validation.go

```diff
@@ -59,9 +60,59 @@ func validateImage(image *api.Image, fldPath *field.Path) field.ErrorList {
 		}
 	}
 
+	for i, sig := range image.Signatures {
+		result = append(result, validateImageSignature(&sig, fldPath.Child("signatures").Index(i))...)
+	}
+
 	return result
 }
 
+func validateImageSignature(signature *api.ImageSignature, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(signature.Type) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("type"), ""))
+	}
+	if len(signature.Content) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("content"), ""))
+	}
+
+	var trustedCondition, forImageCondition *api.SignatureCondition
+	for i := range signature.Conditions {
+		cond := &signature.Conditions[i]
```

Comment:
Because in that way I'd be taking reference to a temporary `cond` value which is reassigned in each iteration. So in the end both `trustedCondition` and `forImageCondition` would point to the same value equal to the last item of `signature.Conditions`.